src/notebooks_service.py

```diff
@@ -229,36 +239,54 @@ def ui(path):
 @authenticated
 def whoami(user):
     """Return information about the authenticated user."""
-    return jsonify(get_user_info(user))
+    user_info = _get_user_info(user)
+    _annotate_servers(user_info['servers'])
+    return jsonify(user_info)
+
+
+def _annotate_servers(servers):
+    """Get servers with renku annotations."""
+    pods = _get_pods().items
+    annotations = {pod.metadata.name: pod.metadata.annotations for pod in pods}
+
+    for server_name, properties in servers.items():
+        pod_annotations = annotations.get(
+            properties.get('state', {}).get('pod_name', {})
+        )
+        servers[server_name]['annotations'] = {
+            key: value
+            for (key, value) in pod_annotations.items()
+            if key.startswith(RENKU_ANNOTATION_PREFIX)
+        }
+    return servers
 
 
 @app.route(urljoin(SERVICE_PREFIX, 'servers'))
 @authenticated
 def user_servers(user):
     """Return a JSON of running servers for the user."""
-    info = get_user_info(user)
-    servers = info['servers']
+    servers = _annotate_servers(_get_user_info(user).get('servers', {}))
+    return jsonify({'servers': servers})
 
-    if KUBERNETES:
-        pods = get_pods().items
-        annotations = {
-            pod.metadata.name: pod.metadata.annotations
-            for pod in pods
-        }
 
-        for server_name, properties in servers.items():
-            pod_annotations = annotations.get(
-                properties.get('state', {}).get('pod_name', {}), {}
-            )
-            servers[server_name]['annotations'] = {
-                key: value
-                for (key, value) in pod_annotations.items()
-                if key.startswith(RENKU_ANNOTATION_PREFIX)
-            }
-    return jsonify({'servers': servers})
+@app.route(
+    urljoin(SERVICE_PREFIX, '<namespace>/<project>/<commit_sha>/logs'),
+    methods=['GET']
+)
+@authenticated
+def server_logs(user, namespace, project, commit_sha):
+    """"Return the logs of the running server."""
+    server = get_user_server(user, namespace, project, commit_sha)
+    if server:
+        # status = SERVER_STATUS_MAP.get(server.get('pending'), 'not found')
+        return v1.read_namespaced_pod_log(
```

Comment:
Should we return `JSON` instead of plain text? This may simplify limiting the number of rows, pagination, ...

Reply:
sure - what should it look like? The k8s api just returns plain text logs, for example

Reply:
This is a log line
```
[D 2019-04-23 08:53:34.263 SingleUserNotebookApp application:555] Looking for jupyter_notebook_config in /home/jovyan/.jupyter
```

* A (possibly) easy solution could be having each line as a string in an array.
```
{
	data: [
		"line 1 plain text",
		"line 2 plain text",
		...
	]
}
```
It's better if we consider a "logical" line, so be careful with multi line logs -- should be easy because it seems they always start with a `tab` instead of a `[`

* A wonderful solution may be having some structured information like
```
{
	data: [
		{
			datetime: "2019-04-23 08:53:34.262",
			source: "application",
			user: "? - SingleUserNotebookApp is not so useful"
			line: 555,
			text: "Looking for jupyter_notebook_config in /home/jovyan/.jupyter"
		},
		{
			...
		},
		...
	]
}
```

Reply:
When I say "single logical line" I mean something like this
```
[D 2019-04-23 08:53:36.998 SingleUserNotebookApp config_manager:98] Paths used for configuration of jupyter_notebook_config: 
    	/etc/jupyter/jupyter_notebook_config.json
```

Reply:
Hmm... This seems like maybe a bit overkill - I need to have a look at how e.g. the docker api does this.

Reply:
We won't always have the same log structure - if, for example rstudio or Matlab are used the logs might be different so we would need to parse them all. One solution would be to read the logs from a logging framework once we deploy one.

Reply:
Ok then, I suggest to go for the a simple solution now (either keeping plain text logs or having a list of strings) and leave further improvements for later.

Reply:
yes, I propose we start with just plaintext and move on from there.

Reply:
Returning plain text is fine, but please make sure to set the corresponding `accept` and `content-type` headers in the request and the response respectively.

Reply:
@ableuler `content-type = text/html` -- should it be something different?

Reply:
is it html? I think `text/plain` would be appropriate here.

Reply:
done